Практическое задание Л10

Gus129 opened the pull request:
Принципы ООП

Gus129 committed:
Package for Lesson_10 created, deleted forked old lesson's packages

Gus129 committed:
Lesson_10 completed

PavelVysotski commented on src/main/java/org/goose/Lesson_10/Task_1/Bowl.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    public void printFoodAmount() {
+        System.out.println("Еды в миске: " +foodAmount);;
```
Пробел после + и два ;

PavelVysotski commented on src/main/java/org/goose/Lesson_10/Task_1/Animal.java:
```diff
@@ -0,0 +1,14 @@
+
+abstract class Animal {
+    protected String name;
+    protected static int animalCount = 0;
```
Как правило, статическое поле делают public, чтобы она была доступна в любом месте проекта. Для этого не нужен метод get... для нее.
Константы с final, зачастую делают private, т.к. во многом она нужна только для конкретного класса, но и public тоже. Короче, все зависит от контекста.
Для статических переменных нет смысла делать метод get.
В остальных классах тоже посмотри.

Gus129 committed:
Task_1 changed access modifiers in all classes. Deleted getters for static variables
Task_2 new interface CalcPerimeterAndArea and class Shape, refactored code to be inline

Gus129 commented on src/main/java/org/goose/Lesson_10/Task_1/Cat.java:
```diff
@@ -39,9 +39,5 @@ public void eat(Bowl bowl) {
     public boolean isFull() {
```
Тут не знаю как поменять видимость чтобы избавиться от этого метода, или вообще не стоит?

PavelVysotski replied:
А что мешает его просто удалить, выпилить из класса?

Gus129 commented on src/main/java/org/goose/Lesson_10/Task_2/Circle.java:
```diff
@@ -1,14 +1,13 @@
         this.radius = radius;
         this.fillColor = fillColor;
         this.borderColor = borderColor;
+        this.shapeType = "Круг";
```
Тут тоже не уверен правильно ли так объявлять

PavelVysotski replied:
Ну это ты верно заметил. Для таких целей используется специальный класс Enum. Почитай, он простой.

Gus129 committed:
Task_1 deleted redundant method isFull()
Task_2 implemented ShapeType as enum
